projects/components/src/overlay/overlay.service.ts

```diff
@@ -37,6 +47,15 @@ export class OverlayService {
       data: config
     });
 
+    popover.closeOnNavigation();
+
+    this.setActivePopover(popover);
+
     return popover;
   }
+
+  private setActivePopover(popover: PopoverRef): void {
+    this.activePopover?.close();
+    this.activePopover = popover;
```

Comment:
this should be unset on popover close

Reply:
Yeah. Is this really a problem though?  I suppose I would have to subscribe to popoverRef.closed$ which I want to avoid.

Reply:
Always hesitant to use subscriptions, but since this is long lived and the subscription is short (and we're holding the popover state anyway), I think subscribing there is just fine. Just becomes a trivial callback of `popover.closed$.subscribe(() => this.clear())` or something

Reply:
not that simple. It should only clear if popover was closed from outside. On line 58, the close call should not reset this.

Reply:
I am thinking this.
```
  private setActivePopover(popover: PopoverRef): void {
    this.closeSubscription?.unsubscribe();
    this.activePopover?.close();

    this.activePopover = popover;
    this.activePopover.closeOnNavigation();
    this.closeSubscription = this.activePopover.closed$.subscribe(() => this.activePopover = undefined);
  }
```

Reply:
why not? line 58 is the case where a new popover is opened without closing the previous one, right? So 58 would be called, the result of which would clear active popover via that subscription. 59 would then execute and set the now undefined value. This could be problematic if closed$ were async, but it's not. If we want to be super careful, we could also just make it to be something like `if (this.activePopover === closedPopover) { this.activePopover = undefined;}`

Reply:
Your suggestion works too, up to you (sorry github really needs to notify me of new comments)